### PR TITLE
[iOS-Sync] Delete Other Devices (Uplift to 1.18.x)

### PR DESCRIPTION
**Pull request by Brandon-T**
## Implementation Details
- Adds support for deleting other devices from the sync chain.
- Adds support for the current device being deleted from the chain, by another device.
- Fixes missing `brave_services_key`

Resolves https://github.com/brave/brave-ios/issues/3194 and https://github.com/brave/brave-ios/issues/3195

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: ios/browser/api/sync/brave_sync_api.mm [hunk 1]
```diff
@@ -180,12 +181,14 @@ - (NSString*)getDeviceListJSON {
...
   base::Value device_list_value(base::Value::Type::LIST);
 
-  for (const auto& device : _worker->GetDeviceList()) {
+  for (const auto& device : device_list) {
     auto device_value = base::Value::FromUniquePtrValue(device->ToValue());
     bool is_current_device =
         local_device_info ? local_device_info->guid() == device->guid() : false;
     device_value.SetBoolKey("isCurrentDevice", is_current_device);
     device_value.SetStringKey("guid", device->guid());
+    device_value.SetBoolKey("supportsSelfDelete",
```
Comment (AlexeyBarabash) [inline ios/browser/api/sync/brave_sync_api.mm:190]: I can't find where `supportsSelfDelete` is used  neither in branch `ios/sync-delete-devices` nor in https://github.com/brave/brave-ios .  Or this is supposed in upcoming PRs?

**Review by AlexeyBarabash — APPROVED** [review]
Looks good

File: ios/browser/api/sync/brave_sync_api.mm (hunk 1 above)
Comment (darkdh) [reply ios/browser/api/sync/brave_sync_api.mm:190]: It would decide whether we allow users to delete remote devices from device list

Comment (AlexeyBarabash) [reply ios/browser/api/sync/brave_sync_api.mm:190]: yeah, so this is for future use?

**Comment by kjozwiak** [comment]
Quick context: Closed as we're going to base `1.23` off of `1.19.x` and not `1.18.x`.

**Pull request closed without merging**